mock/mock.go

```diff
@@ -578,6 +578,19 @@ func (m *Mock) IsMethodCallable(t TestingT, methodName string, arguments ...inte
 	return false
 }
 
+// NumberOfCalls returns the number of times a method was called
+func (m *Mock) NumberOfCalls(methodName string) int {
```

Comment:
This is not actually `NumberOfCalls`, it's more like `NumberOfCallsWithMethodName`. This PR is useful, but do we have to add another method here for people to count calls with certain arguments, or number of calls regardless of method name?

Perhaps something like I mentioned in #1128 is more useful?